runtime/themes/onelight.toml

```diff
@@ -0,0 +1,115 @@
+# One Light
+# Author : erasin<erasinoo@gmail.com>
+
+"attribute" = { fg = "yellow" }
+"comment" = { fg = "gray", modifiers = ["italic"] }
+
+"constant" = { fg = "red" }
+"constant.numeric" = { fg = "gold" }
+"constant.builtin" = { fg = "gold" }
+"constant.character.escape" = { fg = "gold" }
+
+"constructor" = { fg = "blue" }
+
+"function" = { fg = "light-blue" }
+"function.builtin" = { fg = "gray" }
+"function.macro" = { fg = "red" }
+
+"keyword" = { fg = "purple" }
+"keyword.control" = { fg = "purple" }
+"keyword.control.import" = { fg = "purple" }
+"keyword.directive" = { fg = "purple" }
+"keyword.operator" = { fg = "purple" }
+
+"tag" = "yellow"
+"label" = { fg = "cyan" }
+"namespace" = { fg = "red" }
+"operator" = { fg = "red" }
+"property" = { fg = "red" }
+"special" = { fg = "blue" }
+"string" = { fg = "green" }
+"module" = { fg = "cyan" }
+
+"type" = { fg = "yellow" }
+"type.builtin" = "blue"
+
+"punctuation" = "gray"
+"punctuation.delimiter" = "black"
+# "punctuation.bracket" = "black"
+
+# "variable" = { fg = "blue" }
+"variable.builtin" = { fg = "light-blue" }
+"variable.parameter" = { fg = "blue" }
+"variable.other.member" = { fg = "red" }
+
+"markup.heading" = { fg = "red" }
+# "markup.raw" = { bg = "light-white" }
+"markup.raw.inline" = { fg = "green", bg = "light-white" }
+"markup.bold" = { fg = "yellow", modifiers = ["bold"] }
+"markup.italic" = { fg = "purple", modifiers = ["italic"] }
+"markup.list" = { fg = "red" }
+"markup.quote" = { fg = "yellow" }
+"markup.link.url" = { fg = "cyan", modifiers = ["underlined"] }
+"markup.link.text" = { fg = "light-blue" }
+"markup.heading.1" = { fg = "red", modifiers = ["bold"] }
+"markup.heading.2" = { fg = "glod", modifiers = ["bold"] }
+"markup.heading.3" = { fg = "yellow", modifiers = ["bold"] }
+"markup.heading.4" = { fg = "green", modifiers = ["bold"] }
+"markup.heading.5" = { fg = "blue", modifiers = ["bold"] }
+"markup.heading.6" = { fg = "black", modifiers = ["bold"] }
+
+"diff.plus" = "green"
+"diff.delta" = "gold"
+"diff.minus" = "red"
+
+"diagnostic" = { modifiers = ["underlined"] }
+"info" = { fg = "blue", modifiers = ["bold"] }
+"hint" = { fg = "green", modifiers = ["bold"] }
+"warning" = { fg = "yellow", modifiers = ["bold"] }
+"error" = { fg = "red", modifiers = ["bold"] }
+
+"ui.background" = { bg = "white" }
+
+"ui.cursor" = { fg = "black", modifiers = ["reversed"] }
+"ui.cursor.primary" = { fg = "white", bg="black", modifiers = [] }
+"ui.cursor.match" = { fg = "purple", modifiers = ["underlined", "bold"] }
+"ui.cursor.select" = { bg = "purple" }
+"ui.cursor.insert" = { bg = "white" }
+
+"ui.highlight" = { bg = "light-white" }
+
+"ui.selection" = { bg = "light-white" }
+"ui.selection.primary" = { bg = "light-white", modifiers = ["dim"] }
+
+"ui.virtual" = { fg = "light-white"}
+# "ui.virtual.ruler" = {}
+# "ui.virtual.whitespace" = {}
```

Comment:
For these it's typical to highlight `ui.virtual.whitespace` as `{ fg = "light-white" }` but have `ui.virtual.ruler` as `{ bg = "light-white" }`. It's up to you though, a foreground ruler can also make sense if you don't want to see the ruler until characters are on the ruler. What do you think?

(FYI: for testing these, if you're building from latest you can do `:set rulers [80]` to turn on a ruler on column 80 and `:set whitespace.render all` for the visible whitespace).

Reply:
Do not hide virtual characters, but do not hinder the code when use editor.   I haven't used rulers yet, so ...
I change theme file ,add `"ui.virtual.ruler" = { bg = "light-white" }`.